addons/gamification/models/gamification_challenge.py

```diff
@@ -542,12 +542,12 @@ def _get_serialized_challenge_lines(self, user=(), restrict_goals=(), restrict_t
 
             line_data['own_goal_id'] = False,
             line_data['goals'] = []
-            if line.condition=='higher':
-                goals = Goals.search(domain, order="completeness desc, current desc")
-            else:
-                goals = Goals.search(domain, order="completeness desc, current asc")
+            goals = Goals.search(domain)
```

Comment:
The model `gamification.goal` is ordered with `'start_date desc, end_date desc, definition_id, id'`.
This can have a certain performance impact (depending on the number of goals).
In our case, we don't need to have a particular order for the search result.
This can be improved by forcing the order by `id` in the `search` method (with `order='id'`).